pkg/kubelet/cm/cpumanager/metrics_test.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cpumanager
+
+import (
+	"testing"
+
+	"github.com/blang/semver/v4"
+
+	"k8s.io/component-base/metrics/testutil"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpumanager/state"
+	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
+	"k8s.io/kubernetes/pkg/kubelet/metrics"
+	"k8s.io/utils/cpuset"
+)
+
+func TestMetricsAllocate(t *testing.T) {
+	counterMetricValue, _ := testutil.GetCounterMetricValue(metrics.ContainerAlignedComputeResources.WithLabelValues(metrics.AlignScopeContainer, metrics.AlignedPhysicalCPU))
+
+	testCases := []struct {
+		staticPolicyTest
+		expCounterMetricValue float64
+	}{
+		{
+			staticPolicyTest: staticPolicyTest{
+				description: "SocketNoSMT",
+				topo:        topoUncoreDualSocketNoSMT,
+				pod: makeMultiContainerPod(
+					[]struct{ request, limit string }{
+						{"4000m", "4000m"}},
+					[]struct{ request, limit string }{
+						{"2000m", "2000m"}}),
+				containerName:   "initContainer-0",
+				stAssignments:   state.ContainerCPUAssignments{},
+				stDefaultCPUSet: cpuset.New(0, 1, 2, 3, 4, 5, 6, 7),
+			},
+			expCounterMetricValue: counterMetricValue + 2,
+		},
```

Comment:
thanks, LGTM
can we add a tests for:
- ask for odd exclusive CPU count on a non-SMT machine, metric is incremented as expected (that's the point of the fix isn't it?)
- ask for odd exclusive CPU count on SMT machine, we DO NOT increment the metric (non-regression check)

with the tests added I'll add formal LGTM

Reply:
Thanks for your suggestion!

Done, If I got it right ;)